src/transformStateWithClones.js

```diff
@@ -8,6 +8,28 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const newState = [];
```

Comment:
The variable name `newState` is not descriptive of its purpose. Consider renaming it to `stateHistory` or `statesAfterActions` to better reflect that it stores the history of states after each action.